client/script.js

```diff
@@ -244,97 +172,56 @@ function receive(event) {
                 case 0:
-                    loginAlert.innerHTML = "Wrong login or password";
+                    loginAlert.textContent = "Wrong login or password";
                     break;
                 case 1:
-                    loginAlert.innerHTML = "Account with this login already exist";
+                    loginAlert.textContent = "Account with this login already exist";
                     break;
                 case 2:
                     nick = form.login.value;
                     break;
                 case 3:
-                    loginAlert.innerHTML = "This account is already online";
+                    loginAlert.textContent = "This account is already online";
                     break;
                 case 5:
-                    loginAlert.innerHTML = "Stop creating so many accounts!";
+                    loginAlert.textContent = "Stop creating so many accounts!";
                     break;
             }
             break;
         case "token":
-            localStorage.setItem('skrepka.battleships.token.login', rc.data);
+            localStorage.setItem("skrepka.battleships.token.login", rc.data);
             break;
-        case "top": players_top = rc.data; break;
-        case "op_nick": opp_nick = rc.data; break;
-        case "code": code = rc.data; break;
-        }
-        redraw();
-}
-
-
-function redraw() {
-    ctx.clearRect(0, 0, 850, 420);
-    draw_alert();
-    if (ended) {
-        draw_button(field_startX + field_size + 25, field_startY + field_size + 30, "Restart", "#990000");
-    }
-    if (board) {
-        draw_ships(field_startX,field_startY,field_size, board, false);
-    }
-    switch (state) {
-        case -1:
-            draw_top(field_startX+190, field_startY-20);
+        case "top":
+            players_top = rc.data;
             break;
-        case 1:
-            draw_field(field_startX, field_startY);
-            draw_ships_count();
-            draw_button(field_startX + field_size + 70, field_startY + 15, "Auto", "#FFD500");
-            draw_nick();
+        case "op_nick":
+            opp_nick = rc.data;
             break;
-        case 2:
-            draw_field(field_startX, field_startY);
-            draw_top(active_startX+field_size+160, active_startY+40);
-            draw_nick();
-            break;
-        case 3:
-            draw_field(field_startX, field_startY);
-            draw_field(active_startX, active_startY);
-            draw_top(active_startX+field_size+160, active_startY+40);
-            draw_nick();
-            if (opp_board) {
-                draw_ships(active_startX,active_startY, field_size, opp_board, true);
-            }
-            if (board_hits) {
-                draw_ships(field_startX,field_startY, field_size, board_hits, true);
-            }
+        case "code":
+            code = rc.data;
             break;
     }
-}
-
+    redraw();
+};
 
-  function draw_field(stX, stY) {
+const draw_field = (stX, stY) => {
```

Comment:
Is const => better than function?

Reply:
They look cooler. Also they have very small performance benefit because they don't use "this".